static/src/javascripts/projects/common/modules/commercial/creatives/revealer.js

```diff
@@ -24,12 +25,17 @@ define([
                 }
             }).then(function () {
                 return fastdom.read(function () {
-                    return $adSlot[0].getBoundingClientRect();
+                    return detect.getViewport();
+                });
+            }).then(function (viewport) {
+                return fastdom.write(function () {
+                    var background = $adSlot[0].getElementsByClassName('creative__background')[0];
+                    background.style.width = viewport.width + 'px';
+                    // for the height, we need to account for the height of the location bar, which
+                    // may or may not be there. 70px padding is not too much.
+                    background.style.height = (viewport.height + 70) + 'px';
+                    return true;
```

Comment:
Yes, the returned value is important to let our code know the rendering went correctly.